examples/platforms/cc1352/flash.c

```diff
@@ -158,117 +136,84 @@ static void restoreFlashCache(uint32_t mode)
     VIMSLineBufEnable(VIMS_BASE);
 }
 
-/**
- * Translate the errors from the Flash programming FSM to OpenThread error
- * codes.
- *
- * @param [in] error Return from the Flash programming function.
- *
- * @return The corresponding OpenThread @ref otError value.
- */
-static otError fsmErrorToOtError(uint32_t error)
+static uint32_t mapAddress(uint8_t aSwapIndex, uint32_t aOffset)
 {
-    otError ret = OT_ERROR_GENERIC;
+    uint32_t address = FLASH_BASE_ADDRESS + aOffset;
 
-    switch (error)
+    if (aSwapIndex)
     {
-    case FAPI_STATUS_SUCCESS:
-        ret = OT_ERROR_NONE;
-        break;
-
-    case FAPI_STATUS_INCORRECT_DATABUFFER_LENGTH:
-        ret = OT_ERROR_INVALID_ARGS;
-        break;
-
-    case FAPI_STATUS_FSM_ERROR:
-        ret = OT_ERROR_FAILED;
-        break;
-
-    default:
-        break;
+        address += FLASH_SWAP_SIZE;
     }
 
-    return ret;
+    return address;
 }
 
 /**
  * Function documented in platforms/utils/flash.h
  */
-otError utilsFlashInit(void)
+void otPlatFlashInit(otInstance *aInstance)
 {
-    return OT_ERROR_NONE;
+    OT_UNUSED_VARIABLE(aInstance);
 }
 
 /**
  * Function documented in platforms/utils/flash.h
  */
-uint32_t utilsFlashGetSize(void)
+uint32_t otPlatFlashGetSwapSize(otInstance *aInstance)
 {
-    return FlashSizeGet();
+    OT_UNUSED_VARIABLE(aInstance);
+
+    return FLASH_SWAP_SIZE;
 }
 
 /**
  * Function documented in platforms/utils/flash.h
  */
-otError utilsFlashErasePage(uint32_t aAddress)
+void otPlatFlashErase(otInstance *aInstance, uint8_t aSwapIndex)
 {
+    OT_UNUSED_VARIABLE(aInstance);
+
     uint32_t mode;
-    uint32_t fsmRet;
-    otError  ret;
 
-    otEXPECT_ACTION(checkVoltage(), ret = OT_ERROR_FAILED);
+    otEXPECT(checkVoltage());
 
     mode = disableFlashCache();
 
-    fsmRet = FlashSectorErase(aAddress);
+    FlashSectorErase(mapAddress(aSwapIndex, 0));
```

Comment:
This will only erase the page pointed at by `mapAddress`. If `FLASH_PAGE_NUM` is `!=2` this will not erase all necessary pages in the swap index.